patroni/ha.py

```diff
@@ -793,3 +799,17 @@ def run_cycle(self):
         with self._async_executor:
             info = self._run_cycle()
             return (self.is_paused() and 'PAUSE: ' or '') + info
+
+    def watch(self, timeout):
+        cluster = self.cluster
+        # watch on leader key changes if the postgres is running and leader is known and current node is not lock owner
+        if not self._async_executor.busy and cluster and cluster.leader \
+                and cluster.leader.name != self.state_handler.name:
+            leader_index = cluster.leader.index
+        else:
+            leader_index = None
+
+        return self.dcs.watch(leader_index, timeout)
+
+    def wakeup(self):
+        self.dcs.event.set()
```

Comment:
As far as I can tell `wakeup` only works for when the current node is master or running an async action. That should at least be documented. Ideally we should look into interrupting DCS watches on wakeup. I know I will forget `wakeup()` actually means `wakeup_would_be_nice_thankyou_not_a_problem_if_you_cant_right_now()`.